astropy/config/configuration.py

```diff
@@ -594,23 +594,24 @@ def get_config(packageormod=None, reload=False, rootname=None):
         return cobj
 
 
-def generate_config(pkgname='astropy', filename=None):
+def generate_config(pkgname='astropy', filename=None, verbose=False):
     """Generates a configuration file, from the list of `ConfigItem`
     objects for each subpackage.
 
     .. versionadded:: 4.1
 
     Parameters
     ----------
-    packageormod : str or None
+    pkgname : str or None
         The package for which to retrieve the configuration object.
     filename : str or file object or None
         If None, the default configuration path is taken from `get_config`.
 
     """
+    verbosity = nullcontext if verbose else silence
     package = importlib.import_module(pkgname)
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore', AstropyDeprecationWarning)
+    with verbosity(), warnings.catch_warnings():
+        warnings.simplefilter('ignore')
```

Comment:
Can we also only ignore `AstropyDeprecationWarning` (as before) if `verbose=True`?

Reply:
Sure, done.